spinn_front_end_common/abstract_models/abstract_provides_incoming_partition_constraints.py

```diff
@@ -13,12 +13,14 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from six import add_metaclass
 from spinn_utilities.abstract_base import AbstractBase, abstractmethod
+from spinn_utilities.require_subclass import require_subclass
+from pacman.model.graphs import AbstractVertex
 
 
-@add_metaclass(AbstractBase)
-class AbstractProvidesIncomingPartitionConstraints(object):
+@require_subclass(AbstractVertex)
```

Comment:
might want to chekc the interface function for using those. i think its only machine level these days

Reply:
acutally, seems we still support both. maybe we should only support it on 1 though? food for thought